src/lxc/lxccontainer.c

```diff
@@ -924,9 +1027,15 @@ static bool do_lxcapi_start(struct lxc_container *c, int useinit, char * const a
 	}
 	free_init_cmd(init_cmd);
 
-	if (daemonize)
-		exit(ret == 0 ? true : false);
-	return (ret == 0 ? true : false);
+	if (daemonize && ret < 0)
+		exit(EXIT_FAILURE);
+	else if (daemonize)
+		exit(EXIT_SUCCESS);
+
+	if (ret < 0)
+		return false;
```

Comment:
It is not correct here. when got err, we assign `ret = 1`. it should be `if (ret != 0)` or we just delete `ret = 1` when got error.